test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -0,0 +1,94 @@
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "timestamp_manager.h"
+#include <iostream>
+
+timestamp_manager::timestamp_manager() {}
+
+/* Get an instance of connection_simulator class. */
+timestamp_manager &
+timestamp_manager::get_timestamp_manager()
+{
+    static timestamp_manager _timestamp_manager_instance;
+    return (_timestamp_manager_instance);
+}
+
+int
+timestamp_manager::validate_oldest_ts(int new_stable_ts, int new_oldest_ts)
+{
+    connection_simulator *conn = &connection_simulator::get_connection();
+
+    // Oldest timestamp can't move backward.
+    if (new_oldest_ts <= conn->get_oldest_ts()) {
+        std::cout << "Oldest timestamp cannot move backwards." << std::endl;
+        return 1;
+    }
+
+    /*
+     * The oldest and stable timestamps must always satisfy the condition that oldest <= stable.
+     */
+    if (new_oldest_ts > new_stable_ts) {
+        std::cout << "set_timestamp: oldest timestamp " << new_oldest_ts
+                  << " must not be later than stable timestamp " << new_stable_ts << "."
+                  << std::endl;
+        return 1;
+    }
+
+    return 0;
+}
+
+int
+timestamp_manager::validate_stable_ts(int new_stable_ts, int new_oldest_ts)
+{
+    connection_simulator *conn = &connection_simulator::get_connection();
+
+    // Stable timestamp can't move backward.
+    if (new_stable_ts <= conn->get_stable_ts()) {
+        std::cout << "Stable timestamp cannot move backwards." << std::endl;
+        return 1;
+    }
+
+    /*
+     * The oldest and stable timestamps must always satisfy the condition that oldest <= stable.
+     */
+    if (new_oldest_ts > new_stable_ts) {
+        std::cout << "set_timestamp: oldest timestamp " << conn->get_oldest_ts()
```

Comment:
validate_stable_ts will be called when setting the stable timestamp, the error msg indicates that the oldest ts is being set up, need to update the msg to indicate that stable ts cannot be earlier than the oldest ts.

Reply:
Good catch.

Reply:
Although that might be more clear, this is the same error message as what WiredTiger would give in that scenario. Do we want to stay conistent with WT or give clearer message?